packages/xstate-test/test/paths.test.ts

```diff
@@ -255,3 +255,41 @@ describe('getShortestPathsTo', () => {
     expect(openPaths).toHaveLength(1);
   });
 });
+
+describe('getShortestPathsFrom', () => {
+  it('should get shortest paths from array of paths', () => {
+    const machine = createTestMachine({
+      initial: 'a',
+      states: {
+        a: {
+          on: { NEXT: 'b', OTHER: 'b', TO_C: 'c', TO_D: 'd', TO_E: 'e' }
+        },
+        b: {
+          on: {
+            TO_C: 'c',
+            TO_D: 'd'
+          }
+        },
+        c: {},
+        d: {},
+        e: {}
+      }
+    });
+    const model = createTestModel(machine);
+    const pathsToB = model.getShortestPathsTo((state) => state.matches('b'));
+
+    // a (NEXT) -> b
+    // a (OTHER) -> b
+    expect(pathsToB).toHaveLength(2);
+
+    const shortestPaths = model.getShortestPathsFrom(pathsToB);
+
+    // a (NEXT) -> b (TO_C) -> c
+    // a (OTHER) -> b (TO_C) -> c
+    // a (NEXT) -> b (TO_D) -> d
+    // a (OTHER) -> b (TO_D) -> d
+    expect(shortestPaths).toHaveLength(4);
```

Comment:
From this alone it's not completely obvious how this works.

Most existing methods have various flavors and ways to limit path generation. This one only accepts a list of paths (and performs joining under the hood?) but doesn't allow us to do any filtering. It seems that there is a lack of a certain "symmetry" in those APIs

`*From` is not that descriptive here, perhaps `*FromPaths` would work better?

Reply:
What would make this method more symmetrical? Is it the options that are missing?

Reply:
With this PR we have those on the `TestModel`:
```ts
getShortestPaths(options?: Partial<TraversalOptions<TState, TEvent>>): Array<TestPath<TState, TEvent>>;
getPaths(options?: Partial<GetPathsOptions<TState, TEvent>>): Array<TestPath<TState, TEvent>>;
getShortestPathsTo(statePredicate: StatePredicate<TState>): Array<TestPath<TState, TEvent>>;
getSimplePaths(options?: Partial<TraversalOptions<TState, any>>): Array<TestPath<TState, TEvent>>;
getSimplePathsTo(predicate: StatePredicate<TState>): Array<TestPath<TState, TEvent>>;
getPathFromEvents(events: TEvent[], statePredicate: StatePredicate<TState>): TestPath<TState, TEvent>;
getShortestPathsFrom(paths: Array<TestPath<TState, TEvent>>): Array<TestPath<TState, TEvent>>;
```

We can remove `getPathFromEvents` from this set as it seems to be slightly different. However, I'm not quite sure what it does **exactly** because it still calls `performDepthFirstTraversal` internally - so it could also be considered a part of this set of methods, even though its goal might be slightly different.

I see that this added method is not particularly unique in being different but at the very least it's not "symmetric" with others because it doesn't come with a "simple" variant (it only has the "shortest" variant). I think though that it might useful to consolidate stuff further anyway:
1. why does `getShortestPaths` allow options like `eventCases` but `getShortestPathsTo` doesn't? The goal of the latter is to allow for a custom `statePredicate` and not to limit other useful options
2. why do we have `getShortestPaths` and `getPaths`? They both do the same thing **by default**. I think this is a mistake that we should fix. Having aliases like this doesn't bring any real value beyond one of those being shorter and thus maybe it being more visually pleasing. I feel that such aliases come with a real tradeoffs:
- it's bikesheddable for teams which flavour do they want to use
- u need to document both, different articles online might starts using one or another and in the end you create more overall mental overhead for the consumers because of that
- a reader must realize that they are the same or draw the wrong conclusion that they are different "somehow". When I see A and B APIs I won't ever assume that they are merely aliases - I will always assume that there is a meaningful difference between them and I will look for it eventually
- it hurts searchability/"grepability" - it's not uncommon to search the whole codebase for occurrences of a certain API, trying to understand how they are used etc. If we provide aliases and they are both used in the codebase then it makes this job harder to do. 
- I know that `getPaths` actually allow custom `pathGenerator` but I think that in such a case it should **require** it and not default to the shortest paths algorithm (because of the mentioned reasons)

I think that perhaps there is a room for **merging** all of those into one method that would automatically support all of the supported options for all of those flavours~. Of course, some would be defaulted and in such a case it would be OK to default to the shortest paths.

Alternatively, if we want to simplify providing those "standard" path generators then we could have:
- `getShortestPaths(options)`
- `getSimplestPaths(options)`
- `getPaths(optionsWithRequiredPathGenerator)`

Reply:
I completely agree with you regarding `.getPaths(...)`, it should not be an alias and instead require a path generator.

Reply:
Made the path generator required in [49669d0](https://github.com/statelyai/xstate/pull/3446/commits/49669d052d1e2b5b6bcc63b064386c343599c663)

Reply:
1. what does `getPathFromEvents` do?
2. should we introduce `getSimplestPathsFrom` to "match" `getShortestPathsFrom` and bring back the balance to the universe? 😛